### PR TITLE
Refactor `brew cask audit`.

### PR DESCRIPTION
**Pull request by reitermarkus**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

### CODE REVIEW

File: Library/Homebrew/os/mac.rb [hunk 1]
```diff
@@ -68,21 +68,7 @@ def languages
...
-        *os_langs,
-      ].uniq
-
-      # Ensure all languages are valid
-      @languages.select! do |lang|
-        Locale.parse(lang)
-        true
-      rescue Locale::ParserError
-        false
-      end
```
Comment (Bo98) [inline Library/Homebrew/os/mac.rb:83]: If you want to remove this, you'll either need to translate legacy `AppleLangauges` values to modern locales ([it's a long list](https://github.com/apple/swift-corelibs-foundation/blob/42bfad499c4e938156d5b49331c75f69cc92798c/CoreFoundation/Locale.subproj/CFLocaleIdentifier.c#L345-L539)), or adjust the tests [which currently mandate that this returns valid locales](https://github.com/Homebrew/brew/blob/6b2dbbc96f7d8aa12f9b8c9c60107c9cc58befc4/Library/Homebrew/test/os/mac_spec.rb#L8).

Comment (reitermarkus) [reply Library/Homebrew/os/mac.rb:83]: I changed the spec. Parsing to `Locale` is done inside the cask `DSL` anyways.

File: Library/Homebrew/cask/cmd.rb [hunk 2]
```diff
@@ -180,24 +179,38 @@ def self.nice_listing(cask_list)
...
-        true
+      i = 0
+      remaining = []
+
+      while i < all_args.count
+        begin
+          arg = all_args[i]
+
+          remaining << arg unless process_arguments([arg]).empty?
+        rescue OptionParser::MissingArgument
```
Comment (MikeMcQuaid) [inline Library/Homebrew/cask/cmd.rb:203]: Would be good to try and use `cli/parser` here ASAP.

Comment (reitermarkus) [reply Library/Homebrew/cask/cmd.rb:203]: I had a look and it seems like the `cli/parser` will need quite some refactoring in order to support cask subcommands, or rather multiple levels of flags: one for `brew`, `brew cask` and `brew cask <subcommand`.

Comment (MikeMcQuaid) [reply Library/Homebrew/cask/cmd.rb:203]: @reitermarkus Cool. Are you game to start some of that? If not: could you create a help wanted issue to discuss the steps as you see them that are required?

**Pull request merged**